src/math/PercentageMath.sol

```diff
@@ -72,12 +104,41 @@ library PercentageMath {
     /// @param x The value at the start of the interval (included).
     /// @param y The value at the end of the interval (included).
     /// @param percentage The percentage of the interval to be calculated.
-    /// @return The average of x and y, weighted by percentage.
+    /// @return z The average of x and y, weighted by percentage.
     function weightedAvg(
         uint256 x,
         uint256 y,
         uint256 percentage
-    ) internal pure returns (uint256) {
-        return percentMul(x, PERCENTAGE_FACTOR - percentage) + percentMul(y, percentage);
+    ) internal pure returns (uint256 z) {
+        // Let percentage > 0
+        // Underflow if percentage > PERCENTAGE_FACTOR
+        //
+        // Overflow if x1 = x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
+        // <=> x * (PERCENTAGE_FACTOR - percentage) > type(uint256).max - HALF_PERCENTAGE_FACTOR
+        // <=> x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
```

Comment:
`(PERCENTAGE_FACTOR - percentage)` should be greater than 0

Reply:
first condition if not passed then it reverts

Reply:
Not exactly 
There is the case percentage = PERCENTAGE_FACTOR

Reply:
Oh ok understand now what you mean